velox/connectors/hive/HiveConnector.cpp

```diff
@@ -80,29 +85,49 @@ std::string HiveTableHandle::toString() const {
 }
 
 HiveDataSink::HiveDataSink(
-    std::shared_ptr<const RowType> inputType,
-    const std::string& filePath,
-    velox::memory::MemoryPool* memoryPool)
-    : inputType_(inputType) {
+    RowTypePtr inputType,
+    std::shared_ptr<const HiveInsertTableHandle> insertTableHandle,
+    const ConnectorQueryCtx* FOLLY_NONNULL connectorQueryCtx)
+    : inputType_(std::move(inputType)),
+      insertTableHandle_(std::move(insertTableHandle)),
+      connectorQueryCtx_(connectorQueryCtx) {
+  if (!insertTableHandle_->isPartitioned()) {
```

Comment:
Why do we need this logic here? Can we not rely on the logic in HiveDataSink::appendData ?

Reply:
+1

Reply:
The thought is, even when there's no input that triggers HiveDataSink::appendData(), there's still one empty file created as long as a HiveDataSink is instantiated, in the case of unpartitioned table. This is mainly used for the config driver.create_empty_files, combined w/ a piece of logic in TableWriter  https://github.com/facebookincubator/velox/blob/main/velox/exec/TableWriter.cpp#L59-L62